In [ ]:
# %matplotlib widget

from __future__ import annotations

import re
from collections import defaultdict
from dataclasses import dataclass, field
from itertools import permutations, product
from math import inf
from random import choice

import matplotlib.colors as mcolors
import matplotlib.pyplot as plt
import networkx as nx
import numpy as np
import numpy.typing as npt
from mpl_toolkits.mplot3d import axes3d
from numpy import int_, object_
from numpy.typing import NDArray
from test_utilities import test
from util import *

COLORS = list(mcolors.CSS4_COLORS.keys())

<link href="style.css" rel="stylesheet"></link>
<article class="day-desc"><h2>--- Part I---</h2></article>


In [3]:
from typing import Self


tests = [
    {
        "name": "If register C contains 9, the program 2,6 would set register B to 1.",
        "s": """
            Register A: 0
            Register B: 0
            Register C: 9

            Program: 2,6
        """,
        "expected_reg": "B 1",
        "expected": "",
    },
    {
        "name": "If register A contains 10, the program 5,0,5,1,5,4 would output 0,1,2.",
        "s": """
            Register A: 10
            Register B: 0
            Register C: 0

            Program: 5,0,5,1,5,4
        """,
        "expected": "0,1,2",
    },
    {
        "name": "If register A contains 2024, the program 0,1,5,4,3,0 would output 4,2,5,6,7,7,7,7,3,1,0 and leave 0 in register A",
        "s": """
            Register A: 2024
            Register B: 0
            Register C: 0

            Program: 0,1,5,4,3,0
        """,
        "expected_reg": "A 0",
        "expected": "4,2,5,6,7,7,7,7,3,1,0",
    },
    {
        "name": "If register B contains 29, the program 1,7 would set register B to 26.",
        "s": """
            Register A: 0
            Register B: 29
            Register C: 0

            Program: 1,7
        """,
        "expected_reg": "B 26",
        "expected": "",
    },
    {
        "name": "If register B contains 2024 and register C contains 43690, the program 4,0 would set register B to 44354.",
        "s": """
            Register A: 0
            Register B: 2024
            Register C: 43690

            Program: 4,0
        """,
        "expected_reg": "B 44354",
        "expected": "",
    },
    {
        "name": "Example",
        "s": """
            Register A: 729
            Register B: 0
            Register C: 0

            Program: 0,1,5,4,3,0
        """,
        "expected": "4,6,3,5,6,3,5,2,1,0",
    },
]


@test(tests=tests[:])
def partI_test(s: str) -> int:
    pass

TypeError: partI_test() got an unexpected keyword argument 'expected_reg'

In [ ]:
with open("../input/day18.txt") as f:
    puzzle = f.read()

# print(f"Part I: {last(Computer.partI(puzzle))}")

Part I: 7,3,0,5,7,1,4,0,5


<link href="style.css" rel="stylesheet"></link>
<main>

<p>Your puzzle answer was <code>130536</code>.</p><p class="day-success">The first half of this puzzle is complete! It provides one gold star: *</p>


<link href="style.css" rel="stylesheet"></link>
<article class="day-desc"><h2 id="part2">--- Part Two ---</h2><



Test Example passed, for test_partII.
Success


PartII: 202972175280682


<link href="style.css" rel="stylesheet"></link>


<link href="style.css" rel="stylesheet"></link>
<main>

<p>Your puzzle answer was <code>202972175280682</code>.</p><p class="day-success">Both parts of this puzzle are complete! They provide two gold stars: **</p>

</main>
